From 1376a13be77a20e08a622561985e3b8f17715651 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Thu, 17 Mar 2011 13:58:34 +1000
Patch-mainline: 2.6.39
References: fate#310916
Git-commit: c87a8d8dcd2587c203f3dd8a3c5c15d1e128ec0d
Subject: [PATCH 2267/2588] drm/radeon: fixup refcounts in radeon dumb create
 ioctl.

This was using old gem refcounting methods, fix it to be the same
as the normal create ioctl.

Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit c87a8d8dcd2587c203f3dd8a3c5c15d1e128ec0d)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/radeon/radeon_gem.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/radeon/radeon_gem.c b/drivers/gpu/drm/radeon/radeon_gem.c
index 723cd19..aa1ca2d 100644
--- a/drivers/gpu/drm/radeon/radeon_gem.c
+++ b/drivers/gpu/drm/radeon/radeon_gem.c
@@ -358,6 +358,7 @@ int radeon_mode_dumb_create(struct drm_file *file_priv,
 {
 	struct radeon_device *rdev = dev->dev_private;
 	struct drm_gem_object *gobj;
+	uint32_t handle;
 	int r;
 
 	args->pitch = radeon_align_pitch(rdev, args->width, args->bpp, 0) * ((args->bpp + 1) / 8);
@@ -371,12 +372,13 @@ int radeon_mode_dumb_create(struct drm_file *file_priv,
 	if (r)
 		return -ENOMEM;
 
-	r = drm_gem_handle_create(file_priv, gobj, &args->handle);
+	r = drm_gem_handle_create(file_priv, gobj, &handle);
+	/* drop reference from allocate - handle holds it now */
+	drm_gem_object_unreference_unlocked(gobj);
 	if (r) {
-		drm_gem_object_unreference_unlocked(gobj);
 		return r;
 	}
-	drm_gem_object_handle_unreference_unlocked(gobj);
+	args->handle = handle;
 	return 0;
 }
 
-- 
1.7.6

